### PR TITLE
Fix several documentation bugs

### LINKED ISSUE #7227: Add CONTRIBUTING.md
Per guideline:  DO optimize the README.md for the consumer of the client library.

The contributor guide (CONTRIBUTING.md) should be a separate file linked to from the main component README.md.

### LINKED ISSUE #8473: Add service level README
Per https://aka.ms/sdkdocsoverview, slide 7.

### LINKED ISSUE #8651: Update docstrings
- [ ] Standardize doc comment for AddConfigurationSetting
https://review.docs.microsoft.com/en-us/dotnet/api/azure.data.appconfiguration.configurationclient.addconfigurationsetting?view=azure-dotnet-preview&branch=master
- [x] DeleteConfigurationSettingAsync(ConfigurationSetting) is missing a description.
https://review.docs.microsoft.com/en-us/dotnet/api/azure.data.appconfiguration.configurationclient.deleteconfigurationsetting?view=azure-dotnet-preview&branch=master
- [x] "That satisfy the options"
https://review.docs.microsoft.com/en-us/dotnet/api/azure.data.appconfiguration.configurationclient.getrevisionsasync?view=azure-dotnet-preview&branch=master
- [x] Add periods to the end of sentences.
- [ ] Fix "see" in https://review.docs.microsoft.com/en-us/dotnet/api/azure.data.appconfiguration.settingselector?view=azure-dotnet-preview&branch=master

### PR DESCRIPTION
**Pull request by AlexanderSher**
- Fix #7227: Add CONTRIBUTING.md
- Fix #8473: Add service level README
- Fix #8651: Update docstrings

### CODE REVIEW

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs [hunk 1]
```diff
@@ -664,7 +670,7 @@ public virtual AsyncPageable<ConfigurationSetting> GetConfigurationSettingsAsync
         }
 
         /// <summary>
-        /// Retrieves one or more <see cref="ConfigurationSetting"/> that satisfies the options of the <see cref="SettingSelector"/>
+        /// Retrieves one or more <see cref="ConfigurationSetting"/> that satisfy the options of the <see cref="SettingSelector"/>.
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClient.cs]: ConfigurationSetting should be plural here, but it looks funny formatting-wise if we just put an 's' on the end.

What if we re-wrote it to something like "Retrieves one or more `ConfigurationSetting` entities that match the options specified in the passed-in `SettingSelector`" ?

File: sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@
 namespace Azure.Data.AppConfiguration
 {
     /// <summary>
-    /// Options that allow to configure the management of the request sent to the service
+    /// Options that allow to configure the management of the request sent to the service.
```
Comment (annelo-msft) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/src/ConfigurationClientOptions.cs]: Let's change this to "Options that allow users to configure the requests sent to the App Configuration service."

File: sdk/appconfiguration/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# Azure App Configuration client library for .NET
+
+Azure App Configuration is a managed service that helps developers centralize their application and feature settings simply and securely.
+
+- [Azure.Data.AppConfiguration][app_config] allows you to create and manage application configuration settings and retrieve their revisions from a specific point in time
```
Comment (annelo-msft) [inline sdk/appconfiguration/README.md]: End sentence with '.'

**Review by annelo-msft — APPROVED** [review]
Just a few minor comments.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/README.md [hunk 4]
```diff
@@ -212,16 +212,19 @@ Several App Configuration client library samples are available to you in this Gi
...
+This project has adopted the [Microsoft Open Source Code of Conduct][code_of_conduct]. For more information see the [Code of Conduct FAQ][code_of_conduct_faq] or contact [opencode@microsoft.com][email_opencode] with any additional questions or comments.
 
 ![Impressions](https://azure-sdk-impressions.azurewebsites.net/api/impressions/azure-sdk-for-net%2Fsdk%2Fappconfiguration%2FAzure.Data.AppConfiguration%2FREADME.png)
 
 <!-- LINKS -->
 [azconfig_docs]: https://docs.microsoft.com/azure/azure-app-configuration/
+[azconfig_contrib]: ../CONTRIBUTING.md
 [azconfig_setting_concepts]: https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-key-value
 [azconfig_asof_snapshot]: https://docs.microsoft.com/en-us/azure/azure-app-configuration/concept-point-time-snapshot
 [source_root]: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/appconfiguration/Azure.Data.AppConfiguration
```
Comment (weshaggard) [inline sdk/appconfiguration/Azure.Data.AppConfiguration/README.md]: While here can you also fix these links? They should not be pointing directly to master but instead should be relative paths.  See https://github.com/Azure/azure-sdk/issues/798 for more context.

File: sdk/appconfiguration/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+### Samples
+
+Our samples are structured as unit tests so we can easily verify they're up to date and working correctly. These tests aren't recorded and make minimal use of test infrastructure to keep them easy to read.
+
+## Development history
+
+For additional insight and context, the development, release, and issue history for the Azure Event Hubs client library is available in read-only form, in its previous location, the [Azure App Configuration .NET repository]().
+
+<!-- LINKS -->
+[azconfig_root]: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/appconfiguration
```
Comment (weshaggard) [inline sdk/appconfiguration/CONTRIBUTING.md]: Relative paths here as well.

File: sdk/appconfiguration/Azure.Data.AppConfiguration/README.md (hunk 4 above)
Comment (AlexanderSher) [reply sdk/appconfiguration/Azure.Data.AppConfiguration/README.md]: Sure

File: sdk/appconfiguration/CONTRIBUTING.md (hunk 5 above)
Comment (AlexanderSher) [reply sdk/appconfiguration/CONTRIBUTING.md]: Done

**Pull request merged**